### PR TITLE
avoid unused imports

### PR DESCRIPTION
**Pull request by nineninesevenfour**
(second attempt)

fixes https://github.com/eclipse/xtext-xtend/issues/1029

Signed-off-by: Harald Fassler <harald.fassler+9974@gmail.com>

### CODE REVIEW

File: org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -13,7 +13,6 @@ Require-Bundle: org.eclipse.xtext.ide,
  org.antlr.runtime;bundle-version="[3.2.0,3.2.1)"
 Automatic-Module-Name: org.eclipse.xtext.xbase.ide
 Export-Package: .,
- bar,
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: I'm curious: Why did you remove / change the test case?

**Comment by cdietrich** [comment]
i assume we need to adapt the expectations for the last failing test in xtend

Bildschirmfoto 2020-12-11 um 09 21 52
Bildschirmfoto 2020-12-11 um 09 21 43

File: org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF (hunk 1 above)
Comment (nineninesevenfour) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: As far as I understand the purpose of the serialize method that basically means, that only import statements are filtered now, while an a potentially qualification of a reference can still occur.

Comment (nineninesevenfour) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: Are you sure? I removed  org.eclipse.xtext.xbase.testdata/src/bar/String.java

Comment (cdietrich) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: did you forget to push the delete with this PR?

Comment (nineninesevenfour) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: grafik

Comment (cdietrich) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: must be blind. sry

Comment (nineninesevenfour) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: no problem 🙂

**Comment by nineninesevenfour** [comment]
I corrected the expectations in the failing test.
Unfortunately I was not able to run it locally, so I had to make the changes "blindly". Let's wait for the CI builds. 😉
I have to make the test work locally somehow next time.
Thank you both for your patience and help in this. I appreciate it very much! 👍

**Comment by nineninesevenfour** [comment]
The good thing about it: I think I know now, what causes bug 439 and have an idea how to fix it. But I have to intensify my tests clearly.

File: org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF (hunk 1 above)
Comment (nineninesevenfour) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: @szarnekow If you prefer, I can re-add the second test case again, of course.

**Review by cdietrich — APPROVED** [review]

File: org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.xbase.testdata/META-INF/MANIFEST.MF:16]: @nineninesevenfour Yes, that would be good.

**Review by szarnekow — APPROVED** [review]
Deferring the decision whether to create an import clause or not is a good idea. Nice!

**Comment by nineninesevenfour** [comment]
@szarnekow thx! 😊
